tst/com/amazon/corretto/crypto/provider/test/LibCryptoRngTest.java

```diff
@@ -0,0 +1,147 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.amazon.corretto.crypto.provider.test;
+
+import static com.amazon.corretto.crypto.provider.test.TestUtil.sneakyGetInternalClass;
+import static com.amazon.corretto.crypto.provider.test.TestUtil.sneakyInvoke;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import com.amazon.corretto.crypto.provider.LibCryptoRng;
+import com.amazon.corretto.crypto.provider.SelfTestResult;
+import com.amazon.corretto.crypto.provider.SelfTestStatus;
+import org.apache.commons.codec.binary.Hex;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.parallel.Execution;
+import org.junit.jupiter.api.parallel.ExecutionMode;
+import org.junit.jupiter.api.parallel.ResourceAccessMode;
+import org.junit.jupiter.api.parallel.ResourceLock;
+
+@ExtendWith(TestResultLogger.class)
+@Execution(ExecutionMode.CONCURRENT)
+@ResourceLock(value = TestUtil.RESOURCE_GLOBAL, mode = ResourceAccessMode.READ)
+public class LibCryptoRngTest {
+    private LibCryptoRng rnd;
+
+    @BeforeEach
+    public void setup() {
+        rnd = new LibCryptoRng();
+    }
+
+    @AfterEach
+    public void teardown() {
+        rnd = null;
+    }
+
+    // A common mistake it when filling arrays to not do it properly and leave
+    // zero gaps. To detect this, we'll generate arrays of different lengths
+    // and check to see if certain bytes are always zero
+    @Test
+    public void testNextBytes() {
+        for (int size = 0; size < 64; size++) {
+            final byte[] checkArr = new byte[size];
+            final byte[] arr = new byte[size];
+            for (int trial = 0; trial < 4; trial++) {
+                rnd.nextBytes(arr);
+                System.out.println(Hex.encodeHex(arr));
+                for (int x = 0; x < size; x++) {
+                    checkArr[x] = (byte) (checkArr[x] | arr[x]);
+                }
+            }
+            for (int x = 0; x < size; x++) {
+                assertTrue(0 != checkArr[x],
+                        "Check array size " + size + " position " + x + " is equal to zero");
+            }
+        }
+    }
+
+    // A common mistake it when filling arrays to not do it properly and leave
+    // zero gaps. To detect this, we'll generate arrays of different lengths
+    // and check to see if certain bytes are always zero
+    @Test
+    public void testGenerateSeed() {
```

Comment:
I'm going to leave that as-is since I didn't write that test. It's copied from the other DRBG test in ACCP. 
 - https://github.com/corretto/amazon-corretto-crypto-provider/blob/80b3fa918a4787dd5d1d4e3b2451f7e0e5ae5aad/tst/com/amazon/corretto/crypto/provider/test/AesCtrDrbgTest.java#L122